### ISSUE TITLE
[Bug]: Edgecolor doesn't work with pcolormesh

### ISSUE THREAD

**Issue by zxdawn** [matplotlib/matplotlib#21825]
### Bug summary

Edgecolor isn't shown in `pcolomesh` with matplotlib (3.5.0). It works in 3.2.2.

### Code for reproduction

```python
import matplotlib.pyplot as plt 
import numpy as np

Z = np.random.rand(5, 5) 
plt.pcolormesh(Z, cmap='viridis', fc='none', edgecolor='face', lw=5)
```

### Actual outcome

![test_3 5 0](https://user-images.githubusercontent.com/30388627/144217971-78dec972-4a77-403b-b078-41a56be3896d.jpg)

### Expected outcome

![test_3 2 2](https://user-images.githubusercontent.com/30388627/144217986-2f1e849d-ff5d-46ad-b0b0-6a2a71eb92d3.png)

### Operating system

Ubuntu

### Matplotlib Version

3.5.0

### Matplotlib Backend

QtAgg

### Python version

3.8.12

### Installation

conda

**Comment by jklymak** [comment]
I think that was a deliberate fix - if you set the edge color to be the facecolor and you set the facecolor to "none" I'd expect the edgecolor to also be "none".  That this behaved like you see in 3.2.2 I would consider a bug. (probably done in https://github.com/matplotlib/matplotlib/pull/18480)

If you _really_ need this visualization, I could imagine a `edgecolor='mapped'` as a possible value as well to get back the old behaviour.  But it is frankly a unique visualization technique, and perhaps deserving of its own custom method.

**Comment by zxdawn** [comment]
@jklymak If I use `plt.pcolormesh(Z, cmap='viridis', fc='none', edgecolor='mapped', lw=5) `, it will raise this error:
```
~/miniconda3/lib/python3.9/site-packages/matplotlib/colors.py in to_rgba_array(c, alpha)
    357 
    358     if isinstance(c, str):
--> 359         raise ValueError("Using a string of single character colors as "
    360                          "a color sequence is not supported. The colors can "
    361                          "be passed as an explicit list instead.")

ValueError: Using a string of single character colors as a color sequence is not supported. The colors can be passed as an explicit list instead.
```

This visualization is useful when plotting one 2D variable as the backgound and overlaying another 2D variable as edges.

**Comment by jklymak** [comment]
"Mapped" would be a feature request to get you the behaviour you want, not something that already exists.

**Comment by greglucas** [comment]
Note, you can get that effect now by using `None` for edgecolor.
`plt.pcolormesh(Z, cmap='viridis', fc='none', edgecolor=None, lw=5)`

See here for a good example of the options: https://github.com/matplotlib/matplotlib/pull/18480#issuecomment-751126066
Maybe something like that should be added to the docs/examples somewhere similar to the Quadmesh shading argument...

**Comment by zxdawn** [comment]
@greglucas Ha, thanks! The difference between 'none' and None is confusing, hah.

**Comment by greglucas** [comment]
Yes, I agree it is. In matplotlib, `None` is generally taken to meant "use the default value", and `"none"` the string is meant to be "don't draw this".

**Comment by jklymak** [comment]
Hah, thats great @greglucas Thanks!

**Comment by jklymak** [comment]
Sorry, I shouldn't have closed this.  This has a reasonable todo in that there could/should be a defining example of the different combinations.

**Comment by zxdawn** [comment]
I'm not sure whether I find another issue. The facecolor can't be manually changed by `color=(....)`. Here's the question I posted on [StackOverflow](https://stackoverflow.com/q/70541279/7347925).

**Comment by jklymak** [comment]
Yes, that is a different issue.  I'm 95% certain there is an open issue for rgba pcolormesh, but that feature does not currently exist.

**Comment by timhoffm** [comment]
I think this is rightfully closed:

- The original "doesn't work" has been fixed/clarified.
- Given that adjacent edges overdraw each other, I believe the utility of edges is limited (possibly to very smoothly varying data.

Therefore, explaining the details here in an example is acceptable if somebody wants to do the footwork, but we don't need to keep the issue open as desirable workpackage.
